tests/ses/nodes_preparation.pm

```diff
@@ -67,19 +62,47 @@ echo -e '10.0.2.104\tnode4.openqa.test node4' >> /etc/hosts
 EOF
         script_run($_) foreach (split /\n/, $hosts);
         assert_script_run 'cat /etc/hosts';
-        barrier_wait {name => 'network_configured', check_dead_job => $only_master_check};
-        # nodes will ping each other to test connection
-        assert_script_run 'fping -c2 -q $(grep \'openqa.test\' /etc/hosts|awk \'{print$2}\'|tr "\n" " ")';
     }
     else {
-        barrier_wait {name => 'network_configured', check_dead_job => $only_master_check};
-        assert_script_run 'ip a';
-        assert_script_run 'for i in {1..7}; do echo "try $i" && fping -c2 -q updates.suse.com 10.0.2.2 && sleep 2 && break; done';
+        # restart network, get IP from supportserver
+        systemctl 'restart network';
+    }
+    # only one job (master) should check for dead jobs to avoid failures
+    my $only_master_check = check_var('HOSTNAME', 'master') ? 1 : 0;
+    barrier_wait {name => 'network_configured', check_dead_job => $only_master_check};
+    assert_script_run 'ip a';
+    assert_script_run 'for i in {1..7}; do echo "try $i" && fping -c2 -q updates.suse.com && sleep 2 && break; done';
+    # firewall and apparmor should not run
+    my $firewall = is_sle('15+') ? 'firewalld' : 'SuSEfirewall2';
+    systemctl "disable $firewall";
+    systemctl "stop $firewall";
+    systemctl 'disable apparmor';
+    systemctl 'stop apparmor';
+    # chrony is by default not installed on older than SLE 15
+    zypper_call 'in chrony' if is_sle('<15');
+    # configure and start chrony, time synchroniation server for nodes is master
+    my $ntp_server = check_var('HOSTNAME', 'master') ? 'ntp1.suse.de' : 'master.openqa.test';
+    assert_script_run "sed -i '/pool/d' /etc/chrony.conf";
+    if (check_var('HOSTNAME', 'master')) {
+        # set ntp server and add allow for nodes to sync with master
+        assert_script_run "echo -e \"server $ntp_server iburst\\nallow\" >> /etc/chrony.conf";
+        restart_and_sync_chrony;
+        barrier_wait {name => 'master_chrony_ready', check_dead_job => $only_master_check};
+        assert_script_run 'echo "master_minion: master.openqa.test" >/srv/pillar/ceph/master_minion.sls';
+    }
+    else {
+        assert_script_run "echo 'server $ntp_server iburst' >> /etc/chrony.conf";
+        barrier_wait {name => 'master_chrony_ready', check_dead_job => $only_master_check};
+        restart_and_sync_chrony;
+    }
+    assert_script_run "grep -v ^# /etc/chrony.conf";
```

Comment:
Might be cleaner to `upload_logs "/etc/chrony.conf"`, but whatever you prefer..
In case you keep that you might also exclude empty lines.

Reply:
This just for info to see what is in config, no need to upload